controller/comments.go

```diff
@@ -112,10 +113,16 @@ func (c *CommentsController) Delete(ctx *app.DeleteCommentsContext) error {
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		if *identityID != cm.CreatedBy {
-			// need to use the goa.NewErrorClass() func as there is no native support for 403 in goa
-			// and it is not planned to be supported yet: https://github.com/goadesign/goa/pull/1030
-			return jsonapi.JSONErrorResponse(ctx, goa.NewErrorClass("forbidden", 403)("User is not the comment author"))
+		wi, err := appl.WorkItems().LoadByID(ctx.Context, cm.ParentID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		authorized, err := authz.Authorize(ctx, wi.SpaceID.String())
```

Comment:
We can't call authz.Authorize() from a transaction because it may make a remote REST request to Keycloak. We don't want to keep the transaction open for long time.
We will have to have two separate transactions here: load the work item to get the space ID. Then call authz.Authorize() and then a second transaction to delete the comment.